classes/migration/upgrade/v3_5_0/I9425_SeparateUIAndSubmissionLocales.php

```diff
@@ -0,0 +1,96 @@
+
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        /**
+         * Update/add locale arrays
+         */
+
+        $insert = function (object $localeId, string $settingName, string $settingValue): void {
+            DB::table($this->getContextSettingsTable())->insert(
+                [
+                    $this->getContextIdColumn() => $localeId->{$this->getContextIdColumn()},
+                    'locale' => '',
+                    'setting_name' => $settingName,
+                    'setting_value' => $settingValue,
+                ]
+            );
+        };
+        $update = function (object $localeId, string $settingName, string $settingValue): void {
+            DB::table($this->getContextSettingsTable())
+            ->where($this->getContextIdColumn(), '=', $localeId->{$this->getContextIdColumn()})
+            ->where('setting_name', '=', $settingName)
+            ->update(['setting_value' => $settingValue]);
+        };
+        $pluck = fn (object $localeId, string $settingName): array => json_decode(
+            DB::table($this->getContextSettingsTable())
+                ->where($this->getContextIdColumn(), '=', $localeId->{$this->getContextIdColumn()})
+                ->where('setting_name', '=', $settingName)
+                ->pluck('setting_value')[0]
+        );
+        $union = fn (array $a, array $b): string => collect($a)->concat($b)->unique()->sort()->values()->toJson();
+
+        foreach (DB::table($this->getContextTable())->select('primary_locale', $this->getContextIdColumn())->get() as $localeId) {
+            // Add primary locale to form locales
+            $formLocales = $union([$localeId->primary_locale], $pluck($localeId, 'supportedFormLocales'));
+            $update($localeId, 'supportedFormLocales', $formLocales);
+            // Add primary locale to submission locales
+            $submissionLocales = $union([$localeId->primary_locale], $pluck($localeId, 'supportedSubmissionLocales'));
+            $update($localeId, 'supportedSubmissionLocales', $submissionLocales);
+            // supportedDefaultSubmissionLocale from primary locale
+            $insert($localeId, 'supportedDefaultSubmissionLocale', $localeId->primary_locale);
+            // supportedAddedSubmissionLocales from supportedSubmissionLocales and supportedFormLocales
+            $submFormLocales = $union(json_decode($submissionLocales), json_decode($formLocales));
+            $insert($localeId, 'supportedAddedSubmissionLocales', $submFormLocales);
+            // supportedSubmissionMetadataLocales from supportedSubmissionLocales and supportedFormLocales
+            $insert($localeId, 'supportedSubmissionMetadataLocales', $submFormLocales);
+        }
+
+        /**
+         * Update locale character lengths
+         */
+
+        $updateLength = fn (string $l) => collect(DB::select('SELECT DISTINCT TABLE_NAME FROM INFORMATION_SCHEMA.COLUMNS WHERE COLUMN_NAME = ?', [$l]))
```

Comment:
Maybe to also add 
WHERE... AND TABLE_SCHEMA = DB::connection()->getDatabaseName()